plugins/node/opentelemetry-instrumentation-oracledb/src/utils.ts

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright (c) 2025, Oracle and/or its affiliates.
+ * */
+import { safeExecuteInTheMiddle } from '@opentelemetry/instrumentation';
+import {
+  Span,
+  SpanStatusCode,
+  Tracer,
+  context,
+  SpanKind,
+  trace,
+  diag,
+} from '@opentelemetry/api';
+import {
+  SEMATTRS_DB_CONNECTION_STRING,
+  ATTR_SERVER_PORT,
+  ATTR_SERVER_ADDRESS,
+  SEMATTRS_NET_TRANSPORT,
+  SEMATTRS_DB_USER,
+  SEMATTRS_DB_STATEMENT,
+} from '@opentelemetry/semantic-conventions';
+import {
+  ATTR_DB_SYSTEM,
+  ATTR_DB_NAMESPACE,
+  ATTR_DB_OPERATION_NAME,
+} from './semconv';
+import * as oracledbTypes from 'oracledb';
+
+// Local modules.
+import { AttributeNames } from './constants';
+import { OracleInstrumentationConfig, SpanConnectionConfig } from './types';
+import { TraceSpanData, SpanCallLevelConfig } from './internal-types';
+import { SpanNames, DB_SYSTEM_VALUE_ORACLE } from './constants';
+
+const newmoduleExports: any = oracledbTypes;
+
+/**
+ * OracleTelemetryTraceHandler extends TraceHandlerBase from oracledb module
+ * It implements the abstract methods; `onEnterFn`, `onExitFn`, `onBeginRoundTrip`
+ * and `onEndRoundTrip` of TraceHandlerBase class.
+ * Inside these overriden methods, the input traceContext data is used
+ * to generate attributes for span.
+ */
+export class OracleTelemetryTraceHandler extends newmoduleExports.traceHandler
+  .TraceHandlerBase {
+  private _tracer: Tracer;
```

Comment:
the tracer should be updated when `setTracerProvider` is called on the instrumentation class.

Reply:
I am passing the function (`() => this.tracer` ) to OracleTelemetryTraceHandler constructor instead. Just to avoid override of `setTracerProvider` . looks fine?